lib/transport/request.js

```diff
@@ -1,72 +1,107 @@
 'use strict';
 
-const _ = require('lodash');
-const promisifyAll = require('./promiseUtils').promisifyAll;
 const Transport = require('./transport');
-const Request = require('request');
-const dns = require('dns');
+const https = require('node:https');
+// eslint-disable-next-line func-style
+const fetch = (...args) => import('node-fetch').then(({ default: fetch }) => fetch(...args));
 
 const REQUIRED_PROPERTIES = [
   'body',
   'elapsedTime',
   'url',
   'statusCode',
+  'statusText',
   'headers'
 ];
 
-function toAsyncMethod(method) {
-  return `${method.toLowerCase()}Async`;
-}
-
 class RequestTransport extends Transport {
```

Comment:
I meant I'd stand up a FetchTransport in parallel, ie you could also have AxiosTransport, or GotTransport, then configure the default [here](https://github.com/bbc/http-transport/blob/627d4507c49d13ade2fda8fef708fef4bfbdda6b/index.js#L4). That's just an idea, and I can only think a reason for doing to it might be to compare performance. It was more to nod at what the underlying default transport actually was.

Reply:
to add `AxiosTransport` and `GotTransport` would probably be a lot of work and would need to be done as a separate ticket most likely

Reply:
I'm just suggesting that we rename to use `FetchTransport` to indicate what the underlying transport is.